build.ps1

```diff
@@ -131,30 +134,26 @@ Get-ChildItem $PATCHES | where { $_.Name.StartsWith("patch-xmlrpc") -and !$_.Nam
 'dll', 'pdb' | % { "$SCRATCH_DIR\xml-rpc_v45.net\bin\CookComputing.XmlRpcV2." + $_ } |`
   Move-Item -Destination $OUTPUT_45_DIR
 
-#prepare Json.NET 4.8
+#prepare Json.NET 4.5 and 4.8
 
-mkdirClean "$SCRATCH_DIR\json48.net"
-Expand-Archive -DestinationPath "$SCRATCH_DIR\json48.net" -Path "$REPO\Json.NET\Newtonsoft.Json-10.0.2.zip"
-Move-Item "$SCRATCH_DIR\json48.net\Newtonsoft.Json-10.0.2\Src\Newtonsoft.Json" "$SCRATCH_DIR\json48.net"
+mkdirClean "$SCRATCH_DIR\json.net"
+Expand-Archive -DestinationPath "$SCRATCH_DIR\json.net" -Path "$REPO\Json.NET\Newtonsoft.Json-13.0.1.zip"
+Move-Item "$SCRATCH_DIR\json.net\Newtonsoft.Json-13.0.1\Src\Newtonsoft.Json" "$SCRATCH_DIR\json.net"
+Move-Item "$SCRATCH_DIR\json.net\Newtonsoft.Json-13.0.1\Src\NuGet.Config" "$SCRATCH_DIR\json.net"
 
-Get-ChildItem $PATCHES | where { $_.Name.StartsWith("patch-json-net") -and !$_.Name.Contains("dotnet45") } |`
-  % { $_.FullName } | applyPatch -Path "$SCRATCH_DIR\json48.net"
+$RESTORE_NUGET_CONFIG_FILE="$SCRATCH_DIR\json.net\NuGet.Config"
 
-& $msbuild $SWITCHES $FRAME48 $VS2019 $SIGN "$SCRATCH_DIR\json48.net\Newtonsoft.Json\Newtonsoft.Json.Net40.csproj"
-'dll', 'pdb' | % { "$SCRATCH_DIR\json48.net\Newtonsoft.Json\bin\Release\net48\Newtonsoft.Json.CH." + $_ } |`
-  Move-Item -Destination $OUTPUT_48_DIR
-
-#prepare Json.NET 4.5
+((Get-Content -path $RESTORE_NUGET_CONFIG_FILE -Raw) -replace 'https://api.nuget.org/v3/index.json', $NugetSource) |`
+ Set-Content -Path $RESTORE_NUGET_CONFIG_FILE
 
-mkdirClean "$SCRATCH_DIR\json45.net"
-Expand-Archive -DestinationPath "$SCRATCH_DIR\json45.net" -Path "$REPO\Json.NET\Newtonsoft.Json-10.0.2.zip"
-Move-Item "$SCRATCH_DIR\json45.net\Newtonsoft.Json-10.0.2\Src\Newtonsoft.Json" "$SCRATCH_DIR\json45.net"
+Get-ChildItem $PATCHES | where { $_.Name.StartsWith("patch-json-net")} |`
+  % { $_.FullName } | applyPatch -Path "$SCRATCH_DIR\json.net"
 
-Get-ChildItem $PATCHES | where { $_.Name.StartsWith("patch-json-net") -and !$_.Name.Contains("dotnet48") } |`
-  % { $_.FullName } | applyPatch -Path "$SCRATCH_DIR\json45.net"
+& $msbuild $SWITCHES $RESTORE_SWITCHES -p:RestoreConfigFile=$RESTORE_NUGET_CONFIG_FILE $VS2019 $SIGN "$SCRATCH_DIR\json.net\Newtonsoft.Json\Newtonsoft.Json.csproj"
```

Comment:
Suggestion: if you move the definition of `$RESTORE_NUGET_CONFIG_FILE` from line 144 to to line 70, you could compact this to:
`$RESTORE_SWITCHES= '/restore', '/p:RestoreNoCache=true', "/p:RestoreConfigFile=$RESTORE_NUGET_CONFIG_FILE"`

Reply:
I wanted to keep it close to where we unzip the source code and build it for better visibility in case of refactoring